// @Version: proto3
// @File: scenario.proto
// @Version: 1.0.0

syntax = "proto3";

package waymo.open_dataset;

import "compressed_lidar.proto";
import "map.proto";


message ObjectState {
    double center_x = 2;
    double center_y = 3;
    double center_z = 4;

    float length = 5;
    float width = 6;
    float height = 7;

    float heading = 8;

    float velocity_x = 9;
    float velocity_y = 10;

    bool valid = 11;
}

message Track {
    enum ObjectType {
        TYPE_UNSET = 0;
        TYPE_VEHICLE = 1;
        TYPE_PEDESTRIAN = 2;
        TYPE_CYCLIST = 3;
        TYPE_OTHER = 4;
    }

    int32 id = 1;

    ObjectType object_type = 2;

    repeated ObjectState states = 3;
}

message DynamicMapState {
    repeated TrafficSignalLaneState lane_states = 1;
}

message RequiredPrediction {
    enum DifficultyLevel {
        NONE = 0;
        LEVEL_1 = 1;
        LEVEL_2 = 2;
    }

    int32 track_index = 1;

    DifficultyLevel difficulty = 2;
}

message Scenario {
    reserved 9;

    string scenario_id = 5;

    repeated double timestamps_seconds = 1;

    int32 current_time_index = 10;

    repeated Track tracks = 2;

    repeated DynamicMapState dynamic_map_states = 7;

    repeated MapFeature map_features = 8;

    int32 sdc_track_index = 6;

    repeated int32 objects_of_interest = 4;

    repeated RequiredPrediction tracks_to_predict = 11;

    repeated CompressedFrameLaserData compressed_frame_laser_data = 12;
}
